src/couch_mrview/src/couch_mrview_compactor.erl

```diff
@@ -53,8 +53,7 @@ compact(State) ->
         {ok, Fd} = couch_mrview_util:open_file(CompactFName),
         ESt = couch_mrview_util:reset_index(Db, Fd, State),
 
-        {ok, DbReduce} = couch_btree:full_reduce(Db#db.id_tree),
-        Count = element(1, DbReduce),
+        {ok, Count} = couch_db:get_doc_count(Db),
```

Comment:
This could be a problem. The reduce result could be a tuple of either size 2 or 3 (depending on age of view file). get_doc_count uses match so it would crash for tuples of size 2. See https://github.com/cloudant/dbcore/commit/a452a4ab3a46f88a203d0eaac69317e5069f7b1f#diff-0568f6a6952a18b42d6a57157a930333R255. We've found such views in production IRC (@eiri knows more details).